tests/python/contrib/test_hexagon/test_launcher.py

```diff
@@ -40,6 +42,251 @@
 # triggering TIME_WAIT state on the server socket. This prevents another
 # server to bind to the same port until the wait time elapses.
 
+@requires_hexagon_toolchain
+def test_add_hvx(android_serial_number, tvm_tracker_host, tvm_tracker_port, adb_server_socket):
+    """
+    Starting with an elementwise-add computation, try various schedules / optimizations to
+    see the impact they have on performance.
+
+    The main motivation for this test is to explore the relationship between these
+    schedules / optimizations vs. how effectively the primfunc uses the Hexagon's
+    HVX units.
+    """
+
+    host_output_dir = tempfile.mkdtemp()
+
+    print("-"*80)
+    print("OUTPUT DIRECTORY: {}".format(host_output_dir))
+    print("-"*80)
+    print()
+
+    class benchmark_results_collection:
```

Comment:
Can this be pulled out as a global class definition, then exposed as a fixture?  That would allow it to be used/aggregated by all tests, rather than including them all in a single test function.  I'm picturing a session-scope fixture, something like the following.

```python
class BenchmarkResults:
    # Same implementation goes here
    ...

@pytest.fixture(scope="session")
def benchmark_results(benchmark_output_dir):
    results = BenchmarkResults()
    yield results
    
    tabular_output_filename = os.path.join(host_output_dir, 'benchmark-results.csv')
    with open(tabular_output_filename, 'w') as csv_file:
        results.dump(csv_file)
```

Reply:
It's a cool idea.  Right now AFAIK this is the only Hexagon benchmarking script in the neighborhood.  How about I add a "this could be turned into a fixture" commend to the code, and we do that refactoring if/when there's another script that needs it?  Otherwise I fear this would be premature generalization.

Reply:
I think I'd see three advantages, two of which exist even if this is the only.

1. This would separate the execution of the benchmarks from the test case generation separate from the benchmarking framework itself.  Currently, the majority of the benchmarking occurs within 4 nested for-loops, and it isn't immediately obvious on a first read that each case of the benchmarks is independent.

2. Individual benchmarks could be run independently, using pytest's existing command line (e.g. `pytest path/to/benchmark_hexagon.py::test_matmul`)

3. If somebody wants to add a new benchmarked case, it makes it obvious how to add them, by copying the existing benchmark function.  As it is, reading through a 250 line function is possible, but doesn't emphasize where the change would need to be made.

Reply:
Thanks for articulating all that!

I think (1) is somewhat covered by a simple refactoring of the code, as I (currently) understand is advocated in another comment: https://github.com/apache/tvm/pull/10604#discussion_r837572590 

Points (2) and (3) make sense, but AFAICT aren't strictly needed for this script's current requirements.